src/components/Widgets/variants/base/ZapWidget/ZapDepositWidget.tsx

```diff
@@ -84,6 +84,17 @@ export const ZapDepositWidget: FC<ZapDepositWidgetProps> = ({
 
   const widgetConfig = useLiFiWidgetConfig(enhancedCtx);
 
+  // @Note: we want to ensure that we exclude the lp token from possible "Pay With" options [LF-15086]
+  const lpToken = zapData?.market?.lpToken;
+  if (lpToken) {
+    const currentDenyList = widgetConfig.tokens?.deny ?? [];
+    const newDenyList = uniqBy([...currentDenyList, lpToken], 'address');
+
+    widgetConfig.tokens = widgetConfig.tokens ?? {};
+    widgetConfig.tokens.deny = widgetConfig.tokens.deny ?? [];
+    widgetConfig.tokens.deny = newDenyList;
```

Comment:
We could directly assign the updated value to the `widgetConfig.tokens`, but it's more of a nit-pick 😬 
```suggestion
    widgetConfig.tokens = {
    ...(widgetConfig.tokens ?? {}),
    deny: uniqBy(
      [...(widgetConfig.tokens?.deny ?? []), lpToken],
      'address'
    ),
  };
```

Reply:
You could also use lodash and get rid of most defaults & question marks:

```ts
    update(widgetConfig, 'tokens.deny', (current) => {
      return uniqBy([...(current ?? []), lpToken], 'address');
    });
```

The three are functionally equivalent, so it's really about what's more maintainable,

I'd find it easier to maintain the first one to be honest ^^'
Let me know if you feel strongly about using the other version.

Reply:
No, fair, let's go with the first one and we'll adjust if needed 💯